ocaml/xapi/xapi_systemctl.ml

```diff
@@ -0,0 +1,51 @@
+(*
+ * Copyright (C) 2006-2009 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+(**
+ * @system service management
+*)
+
+module D = Debug.Make (struct let name = "xapi_systemctl" end)
+
+open D
+
+type t = Start | Stop
+
+let to_string = function Start -> "start" | Stop -> "stop"
+
+let operate ~wait_until_success ~service ~timeout op =
+  let op_str = op |> to_string in
+  try
+    debug "%s %s" op_str service ;
+    ignore
+      (Forkhelpers.execute_command_get_output !Xapi_globs.systemctl
```

Comment:
Fe_systemctl is not used here because it creates and deletes the service files on start and stop, something that's is not convenient to use here.

There are a few non-transient services managed from xapi, so there's an oportunity to consolidate these systemctl calls.